integration-tests/utils/integration-utils.js

```diff
@@ -106,7 +106,18 @@ const clickTableCell = async (page, { tableName, columnHeader, text, isDescendan
 }
 
 const click = async (page, xpath, options) => {
-  return (await page.waitForXPath(xpath, defaultToVisibleTrue(options))).click()
+  try {
+    return (await page.waitForXPath(xpath, defaultToVisibleTrue(options))).click()
+  } catch (e) {
+    // Handle Error: Node is detached from document
+    //   at ElementHandle._scrollIntoViewIfNeeded (/mnt/ramdisk/.yarn/unplugged/puppeteer-npm-13.5.1-231e586c7a/node_modules/puppeteer/src/common/JSHandle.ts:475:22)
+    //   at processTicksAndRejections (node:internal/process/task_queues:96:5)
+    //   at ElementHandle.click (/mnt/ramdisk/.yarn/unplugged/puppeteer-npm-13.5.1-231e586c7a/node_modules/puppeteer/src/common/JSHandle.ts:617:5)
+    if (e instanceof Error && e.message?.includes('Node is detached from document')) {
```

Comment:
Is this the right way to check error message?

Reply:
It's probably safe to assume that `e` is an instance of Error here. Ideally, it's better to check errors based on an error code or something like that instead of a user facing message. That way, you avoid the risk of code breaking if the message is slightly reworded. Not all libraries provide that option though. It may be worth checking if Puppeteer uses a specific error class for this type of error ([puppeteer.errors](https://pptr.dev/api/puppeteer.errors)).

Reply:
Fair enough. Yeah, there's no guarantee that a library will always provide an error code or similar. Sometimes the only option is to look at the message.